service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/support/MigrateCaseController.java

```diff
@@ -62,10 +63,50 @@ public class MigrateCaseController extends CallbackController {
+        "DFPL-CFV", this::runCFV,
+        "DFPL-CFV-Rollback", this::runCFVrollback
     );
 
+    private void runCFV(CaseDetails caseDetails) {
+        CaseData caseData = getCaseData(caseDetails);
+        migrateCaseService.doHasCFVMigratedCheck(caseDetails.getId(), (String) caseDetails.getData()
+                .get("hasBeenCFVMigrated"), "DFPL-CFV");
+        caseDetails.getData().putAll(migrateCaseService.migrateApplicantWitnessStatements(caseData));
+        caseDetails.getData().putAll(migrateCaseService
+            .migrateApplicationDocumentsToDocumentsFiledOnIssueList(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateApplicationDocumentsToThresholdList(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateApplicationDocumentsToCarePlanList(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateCourtBundle(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateCorrespondenceDocuments(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateExpertReports(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateGuardianReports(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateNoticeOfActingOrIssue(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migratePositionStatementRespondent(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migratePositionStatementChild(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateRespondentStatement(caseData));
+        caseDetails.getData().putAll(migrateCaseService.migrateSkeletonArgumentList(caseData));
+        caseDetails.getData().putAll(migrateCaseService
+            .moveCaseSummaryWithConfidentialAddressToCaseSummaryListLA(caseData));
+        caseDetails.getData().put("hasBeenCFVMigrated", YesNo.YES);
+    }
+
+    private void runCFVrollback(CaseDetails caseDetails) {
+        migrateCaseService.doHasCFVMigratedCheck(caseDetails.getId(), (String) caseDetails.getData()
+                .get("hasBeenCFVMigrated"), "DFPL-CFV", true);
+        migrateCaseService.rollbackApplicantWitnessStatements(caseDetails);
+        migrateCaseService.rollbackApplicationDocuments(caseDetails);
+        migrateCaseService.rollbackCaseSummaryMigration(caseDetails);
+        migrateCaseService.rollbackCourtBundleMigration(caseDetails);
+        migrateCaseService.rollbackCorrespondenceDocuments(caseDetails);
+        migrateCaseService.rollbackExpertReports(caseDetails);
+        migrateCaseService.rollbackGuardianReports(caseDetails);
+        migrateCaseService.rollbackNoticeOfActingOrIssue(caseDetails);
+        migrateCaseService.rollbackPositionStatementChild(caseDetails);
+        migrateCaseService.rollbackPositionStatementRespondent(caseDetails);
+        migrateCaseService.rollbackSkeletonArgumentList(caseDetails);
+        caseDetails.getData().put("hasBeenCFVMigrated", YesNo.NO);
```

Comment:
`caseDetails.getData().remove` might be better here, as we want all new fields to be rolled back and that includes this one!